docs/misc/new_design.md

```diff
@@ -0,0 +1,169 @@
+# Rethinking the Design of Mathematical Theory Development in Lean
+
+It turns out that we must have a design template in mind in order to carry on further designing of GA in Lean. This document will address this issue.
+
+## Key Insight from Euclidean Geometry
+
+Taking a step back, let's go back to geometry and look at https://github.com/vaibhavkarve/leanteach2020/blob/master/src/euclid.lean .
+
+What's a point? How do we tell this concept to the computer? Do we draw a point? Or do we went to analytic geometry and give it a coordinate? It turns out we don't need to do either, the former is infeasible and the latter is the worst idea in formalizing.
+
+```lean
+constant Point : Type
```

Comment:
What if I have a `pointXY` structure, and I want to apply the theorems about `Point` to it?

Reply:
This paragraph discuss how to do abstract reasoning without ever coming back to the concrete world. There was supposed to be a paragraph about what if we need to link back to the concrete world and that's where typeclasses coming. And I should also discuss using class v.s. structure (1:1 v.s. 1:n) as pointed out by Kevin Buzzard.